x/evm/keeper/state_transition.go

```diff
@@ -140,14 +144,27 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 	txHash := tx.Hash()
 	res.Hash = txHash.Hex()
 
-	logs := k.GetTxLogs(txHash)
-	res.Logs = types.NewLogsFromEth(logs)
+	// Set the bloom filter and commit only if transaction is NOT reverted
+	if !res.Reverted {
+		logs := k.GetTxLogs(txHash)
+		res.Logs = types.NewLogsFromEth(logs)
+		// update block bloom filter
+		bloom := k.GetBlockBloomTransient()
+		bloom.Or(bloom, big.NewInt(0).SetBytes(ethtypes.LogsBloom(logs)))
+		k.SetBlockBloomTransient(bloom)
 
-	// update block bloom filter
-	bloom := k.GetBlockBloomTransient()
-	bloom.Or(bloom, big.NewInt(0).SetBytes(ethtypes.LogsBloom(logs)))
-	k.SetBlockBloomTransient(bloom)
+		commit()
```

Comment:
wondering if we also need the logs on the reverted response? What's your thought of using this approach:

```go
// Set the logs to response
logs := k.GetTxLogs(txHash)
res.Logs = types.NewLogsFromEth(logs)

// update block bloom filter
bloom := k.GetBlockBloomTransient()
bloom.Or(bloom, big.NewInt(0).SetBytes(ethtypes.LogsBloom(logs)))
k.SetBlockBloomTransient(bloom)

// commit only if transaction is NOT reverted
if !res.Reverted {
    commit()
}
```

Reply:
I think log is not needed in case evm revert. @yijiasu-crypto can you confirm?

Reply:
log is not needed. When evm revert occurs there's no logs at all

Reply:
No, when a revert occurs, the transaction state changes are thrown away, including any event logs.

This behaviour is defined in the Yellow Paper here:

![image](https://user-images.githubusercontent.com/85918228/124859213-0f771f00-dfe2-11eb-8465-6831d946125f.png)

The A0 bit when w = REVERT is the formal version of what I said above: no logs are recorded in the case of a revert.

Ref: https://ethereum.stackexchange.com/questions/38019/is-it-possible-to-retrieve-an-event-log-from-a-reverted-transaction